File: baseadapter/src/main/java/com/sum/simple/base/ViewHolder.java
package com.sum.simple.base;

import androidx.recyclerview.widget.RecyclerView;
import androidx.viewbinding.ViewBinding;

/**
 * @author liujiang
 * Desc:
 */
public class ViewHolder<V extends ViewBinding> extends RecyclerView.ViewHolder {
    public V binding;

    public ViewHolder(V item) {
        super(item.getRoot());
        binding = item;
    }
}

File: sampleAdapter/src/main/java/com/sum/sampleadapter/ViewHolder.kt
package com.sum.sampleadapter

import androidx.recyclerview.widget.RecyclerView
import androidx.viewbinding.ViewBinding

/**
 * @author  LiuJiang
 * Desc:
 */
class ViewHolder<V: ViewBinding>(val binding: V) : RecyclerView.ViewHolder(binding.root) {
}